connectors/sources/tests/test_generic_database.py

```diff
@@ -130,37 +130,40 @@ def test_get_configuration(patch_logger):
     assert config["host"] == "127.0.0.1"
 
 
-def test_validate_configuration_missing_fields(patch_logger):
-    """Test _validate_configuration method check missing fields"""
+@pytest.mark.asyncio
+async def test_validate_config_missing_fields(patch_logger):
+    """Test validate_config method check missing fields"""
     # Setup
     source = create_source(GenericBaseDataSource)
     with pytest.raises(Exception):
         source.configuration.set_field(name="host", value="")
 
         # Execute
-        source._validate_configuration()
+        await source.validate_config()
 
 
-def test_validate_configuration_port(patch_logger):
-    """Test _validate_configuration method check port"""
+@pytest.mark.asyncio
+async def test_validate_config_port(patch_logger):
+    """Test validate_config method check port"""
     # Setup
     source = create_source(GenericBaseDataSource)
     with pytest.raises(Exception):
         source.configuration.set_field(name="port", value="abcd")
 
         # Execute
-        source._validate_configuration()
+        await source.validate_config()
 
 
-def test_validate_configuration_ssl(patch_logger):
-    """Test _validate_configuration method check port"""
+@pytest.mark.asyncio
+async def test_validate_config_ssl(patch_logger):
+    """Test validate_config method check ssl"""
     # Setup
     source = create_source(PostgreSQLDataSource)
     source.configuration.set_field(name="ssl_disabled", value=False)
 
     with pytest.raises(Exception):
         # Execute
-        source._validate_configuration()
+        await source.validate_config()
```

Comment:
Is there also a positive test case somewhere, i.e. the configuration is valid and no exception is raised?

Reply:
Done